e2e/README.md

```diff
@@ -0,0 +1,108 @@
+# E2E performance regression tests
+
+This directory contains the scripts and configuration files for running the
+performance regression tests. These tests are called E2E tests because they
+run the app on a real device (physical or emulated).
+
+![Example of a e2e test run](./example.gif);
+
+To run the e2e tests:
+
+1. Connect an android device. The tests are currently designed to run only on android. It can be
+   a physical device or an emulator.
+
+2. Make sure Fastlane was initialized by running `bundle install`
+
+3. Run the tests with `npm run test:e2e`.
+
+## Performance regression testing
+
+The output of the tests is a set of performance metrics (see video above).
+The test will tell you if the performance significantly worsened for any test case.
+
+For this to work you need a baseline you test against. The baseline is set by default
+to the `main` branch.
+
+The test suite will run each test-case twice, once on the baseline, and then on the branch
+you are currently on.
+
+It will run the tests of a test case multiple time to average out the results.
+
+## Structure
+
+For the test suite, no additional tooling was used. It is made of the following
+components:
+
+- Test server:
+  - A nodeJS application that starts an HTTP server.
+  - Receives test results from the app.
+  - Located in `e2e/server`.
+
+- Client:
+  - Client-side code (app) for communication with the test server.
+  - Located in `src/libs/E2E/client.js`.
+
+- The tests themselves :
+  - Currently, there is only one test case, which test the performance of the app start time
+  - The tests are _inside the app_, and execute logic using app code (e.g. `navigationRef.navigate('Signin')`)
+  - Located inside the app `src/libs/E2E/reactNativeLaunchingTest.js`
+
+- The test runner:
+    - Orchestrates the test suite.
+    - Runs the app with the tests on a device
+    - Responsible for gathering and comparing results
+    - Located in `e2e/testRunner.js`.
+
+
+## How a test gets executed
+
+There exists a custom android entry point for the app, which is used for the e2e tests.
+The entry file used is `src/libs/E2E/reactNativeEntry.js`, and here we can add our test case.
+
+The test case should only execute its test once. The _test runner_ is responsible for running the
+test multiple time to average out the results.
```

Comment:
Does it mean that we will run the same test several times on CI? 
How much time it takes to run it?
I now see why you can use z-score (https://en.wikipedia.org/wiki/Central_limit_theorem) but I guess it can take some time.  How many times do we repeat that test?

Reply:
In reassure the tests are quite fast so I guess they can do them several times but can we do it too?

Reply:
Another option might be to run the app and tests on iOS as its more stable in producing results

Reply:
We need android as it's usually much slower

Reply:
Let me think about it more

Reply:
I think it's definitely good starting point and already useful even if it takes an hour to run

Reply:
Is it free?

Reply:
yeah, i double checked that with rory and its actually free (super nice move by github)